docassemble/ALAffidavitOfIndigency/data/questions/affidavit_body.yml

```diff
@@ -169,19 +178,20 @@ buttons:
   - Exit: exit
     url: https://massaccess.suffolklitlab.org
 ---
-reconsider: True
+depends on:
+  - additional_income_allowed
+  - household_monthly_income
 code: |
-  additional_income_allowed = household_size * poverty_increment
+  additional_income_allowed = max(household_size() - 1, 0) * poverty_increment
   household_income_limit = (poverty_base + additional_income_allowed) * poverty_multiplier
 
-  household_income_qualifies = int((household_income_limit)/12) >=  int(household_monthly_income)
----
-reconsider: True
-code: |
-  household_size = household_additional_size + 1 if has_household_members else 1
+  household_income_qualifies = int((household_income_limit)/12) >=  int(hh_income.amount(period_to_use=12))
 ---
 code: |
-  household_monthly_income = hh_income.amount(period_to_use=12)
+  def household_size():
+    if public_assistance_kinds.any_true():
+      return DAEmpty()
+    return household_additional_size + 1 if has_household_members else 1
```

Comment:
This is pretty confusing

Reply:
I removed it for now, but I've been running into a lot of limits of `depends on` and `on change` and making variables into functions for simple calculations seems to be the best:
* `depends on` and `on change` also don't play well with `skip undefined: True`, just like `reconsider: True` does. The attachment block will just output blanks for variables that were undefined
* `depends on` and `on change` also don't seem to play well with lists and gathered attributes changing: variables like `users` don't properly trigger code blocks, I'd guess because DA is just checking variable references, and not all of their elements and attributes. There isn't a reliable way to recalculate those variables: you'd have to list all of the attributes that could possibly cause a variable to change. For example, `case_name` would need to explicitly list:   
    * `users[0].name.first`
    * `users[0].name.middle`
    * `users[0].name.last`
    * `users[1].name.first`
    * ... and so on

Reply:
The dual responsibility of the function was part of the problem--I can remember why I made the value a `DAEmpty` as a quick fix, but it's weird to encapsulate that behavior. Probably best to put that logic in the interview order block and attachment block instead, or rely on `skip undefined: True` at a minimum, but I originally went for a quick fix without knowing all possible solutions (this was originally built in the first weeks of lockdown when getting a remote signature on a fee waiver was important for e-filing, because e-filing also had a per envelope charge).

Reply:
Gotcha. I think it's less confusing now, it's still necessary, just because there are quite a few screens that use it. Can't quite simplify as much as I want.